api/app/GraphQL/Mutations/UpdatePoolCandidateStatus.php

```diff
@@ -12,41 +12,58 @@
     public function __invoke(null $_, array $args)
     {
         $candidate = PoolCandidate::findOrFail($args['id']);
+        $values = [];
 
         if (isset($args['pool_candidate_status']) && $args['pool_candidate_status'] !== $candidate->pool_candidate_status) {
             $status = $args['pool_candidate_status'];
 
-            $timestamp = match ($status) {
+            $now = now();
+            $values = [
+                'removed_at' => null,
+                'final_decision_at' => null,
+                'placed_at' => null,
+            ];
+
+            $timestamps = match ($status) {
+                PoolCandidateStatus::EXPIRED->name,
                 PoolCandidateStatus::SCREENED_OUT_ASSESSMENT->name,
                 PoolCandidateStatus::SCREENED_OUT_APPLICATION->name,
-                PoolCandidateStatus::QUALIFIED_AVAILABLE->name => 'final_decision_at',
+                PoolCandidateStatus::QUALIFIED_UNAVAILABLE->name,
+                PoolCandidateStatus::QUALIFIED_WITHDREW->name,
+                PoolCandidateStatus::QUALIFIED_AVAILABLE->name => ['final_decision_at'],
 
                 PoolCandidateStatus::SCREENED_OUT_NOT_RESPONSIVE->name,
                 PoolCandidateStatus::SCREENED_OUT_NOT_INTERESTED->name,
-                PoolCandidateStatus::QUALIFIED_UNAVAILABLE->name,
-                PoolCandidateStatus::QUALIFIED_WITHDREW->name,
-                PoolCandidateStatus::REMOVED->name => 'removed_at',
+                PoolCandidateStatus::REMOVED->name => ['removed_at'],
 
                 PoolCandidateStatus::PLACED_TENTATIVE->name,
                 PoolCandidateStatus::PLACED_CASUAL->name,
                 PoolCandidateStatus::PLACED_TERM->name,
-                PoolCandidateStatus::PLACED_INDETERMINATE->name => 'placed_at',
+                PoolCandidateStatus::PLACED_INDETERMINATE->name => ['placed_at', 'final_decision_at'],
 
                 default => null// no-op
             };
 
-            if ($timestamp) {
-                $candidate->$timestamp = now();
+            if ($timestamps) {
+                foreach ($timestamps as $timestamp) {
+                    $values[$timestamp] = $now;
```

Comment:
I think if a timestamp is already set it would make sense to leave it with it's current value

Reply:
Hmm, I could be misunderstanding but the bug we are fixing is because we were not unsetting timestamps when they needed to so the intention is to fix that by unsetting those that are set when they should not be.

Reply:
The goal of the issue is to ensure timestamps are null or not based on the status. I'm saying here that if we want the status to be set and it already is we could probably leave it alone.

For example if someone set a candidate to QUALIFIED_AVAILABLE a week ago their finalDecisionAt will be set to then. If we then set the user to placed casual we need to set placedAt to now but we could probably leave their finalDecisionAt as the date a week ago.

Reply:
OH! okay, yeah I was just misunderstanding. That makes sense.

Reply:
Actually @petertgiles this is a good question. If they are changing a "final" decision to a different "final" decision should we record it as the last time the decision was set or leave the original.

Sadly, the "final" portion is not 100% true so it technically makes sense both ways :sweat:

Reply:
It's not super obvious, is it?  :smile:  I think resetting a previously set field to "now" on a status change makes sense, like the "updated at" field.  If we need a log of the times all changes we should be referring to activity_log anyway.

Reply:
That argument works for me.